chalicelib/action.py

```diff
@@ -32,46 +40,210 @@ def __init__(self, payload, config):
+
+        for event in data:
+            event_date = event.get("event_date")
+            reason = event.get("reason")
+            hours = event.get("hours")
+            self.slack.add_section_block(
+                text=f"Date: *{event_date}*\nReason: *{reason}*\nHours: *{hours}*"
+            )
+            self.slack.add_divider_block()
+
+    def _check_locks(self, date: datetime, second_date: datetime) -> list:
+        """
+        Check dates for lock.
+        """
+        dates_to_check = list()
+        locked_dates = list()
+        for date in date_range(start_date=date, stop_date=second_date):
+            if not date.strftime("%Y-%m") in dates_to_check:
+                dates_to_check.append(date.strftime("%Y-%m"))
+
+        log.debug(f"Got {len(dates_to_check)} date(s) to check")
+        for date in dates_to_check:
+            response = read_lock(
+                url=self.config["backend_url"], user_id=self.user_id, date=date
+            )
+            if response.json():
+                log.info(f"Date {date} is locked")
+                dates_to_check.append(response.json())
+                locked_dates.append(date)
+
+        return locked_dates
+
+    def _valid_reason(self, reason: str) -> bool:
+        """
+        Check that reason is valid
+        """
+
+        if reason in self.config.get("valid_reasons"):
+            return True
+
+        return False
+
+    def _valid_number_of_args(self, min_args: int, max_args: int) -> bool:
+        """
+        Check that the number of arguments in the list is within the valid range
+        """
+        log.debug(f"Got {len(self.arguments)} number of args")
+        if min_args <= len(self.arguments) <= max_args:
+            return True
+        return False
+
     def perform_action(self):
         """
-        Perform action.
-
-        Supported actions are:
-        add - Add one or more events in timereport
-        edit - Edit a single event in timereport
-        delete - Delete event in timereport
-        list - List one or more events in timereport
-        lock - Lock month to show you're done with the reporting
-        help - Provide this helpful output
-        help <action> - Provide helpful output for specific action
+        Run the action
         """
+        raise NotImplementedError()
 
-        self.action = self.params[0]
-        self.arguments = self.params[1:]
-        self.user_id = self.payload["user_id"][0]
-        self.user_name = self.payload["user_name"][0]
 
-        log.debug(f"Action is: {self.action}")
-        log.debug(f"Arguments are: {self.arguments}")
+class HelpAction(BaseAction):
+    name = "help"
+    short_doc = "Provide this helpful output"
+    doc = """
+        Nothing to see here.. Try another action :)
+        """
 
-        action_fn = self.actions.get(self.action, self._unsupported_action)
-        return action_fn()
+    def perform_action(self):
+        if len(self.arguments) > 0:
```

Comment:
If the input is `/timereport help foo` this will not return anything to the user, since it will never enter the else statement